pkg/key/catalog.go

```diff
@@ -0,0 +1,60 @@
+package key
+
+import (
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/k8smetadata/pkg/label"
+	"github.com/giantswarm/microerror"
+)
+
+func CatalogTitle(customResource v1alpha1.Catalog) string {
+	return customResource.Spec.Title
+}
+
+func CatalogStorageURL(customResource v1alpha1.Catalog) string {
+	return customResource.Spec.Storage.URL
+}
+
+func CatalogConfigMapName(customResource v1alpha1.Catalog) string {
+	return customResource.Spec.Config.ConfigMap.Name
```

Comment:
The config struct and fields below are optional. So should we add nil checks?

Components using the key funcs then don't need to worry about this. WDYT?

Reply:
That's nice, I think we can return an empty string in that case, right? 
I will do that :) Thanks for your suggestion!

Reply:
Yes, we can return empty strings. Thx!